src/agent/debuglet.ts

```diff
@@ -198,107 +200,100 @@ export class Debuglet extends EventEmitter {
-                          err4);
-                      that.emit('initError', err4);
-                      return;
-                    }
-
-                    that.getSourceContext_(function(err5, sourceContext) {
-                      if (err5) {
-                        that.logger_.warn(
-                            'Unable to discover source context', err5);
-                        // This is ignorable.
-                      }
+    // TODO: Verify that it is fine for `id` to be undefined.
+    let id: string|undefined;
+    if (process.env.GAE_MINOR_VERSION) {
+      id = 'GAE-' + process.env.GAE_MINOR_VERSION;
+    }
 
-                      if (semver.satisfies(process.version, '5.2 || <4')) {
-                        // Using an unsupported version. We report an error
-                        // message about the Node.js version, but we keep on
-                        // running. The idea is that the user may miss the error
-                        // message on the console. This way we can report the
-                        // error when the user tries to set a breakpoint.
-                        that.logger_.error(NODE_VERSION_MESSAGE);
-                      }
+    let fileStats: scanner.ScanResults;
+    try {
+      // TODO: Address the case when `that.config_.workingDirectory` is
+      //       `null`.
+      fileStats = await scanner.scan(
+          !id, that.config_.workingDirectory as string, /.js$|.map$/);
+    } catch (err) {
+      that.logger_.error('Error scanning the filesystem.', err);
+      that.emit('initError', err);
+      return;
+    }
 
-                      // We can register as a debuggee now.
-                      that.logger_.debug('Starting debuggee, project', project);
-                      that.running_ = true;
-                      // TODO: Address the case where `project` is `undefined`.
-                      that.project_ = project as string;
-                      that.debuggee_ = Debuglet.createDebuggee(
-                          // TODO: Address the case when `project` is
-                          // `undefined`.
-                          // TODO: Address the case when `id` is `undefined`.
-                          project as string, id as string,
-                          that.config_.serviceContext,
-                          // TODO: Handle the case where `onGCP` is `undefined`.
-                          sourceContext, that.config_.description, null,
-                          onGCP as boolean);
-                      that.scheduleRegistration_(0 /* immediately */);
-                      that.emit('started');
-                    });
-                  });
-                });
-              });
+    const jsStats = fileStats.selectStats(/.js$/);
+    const mapFiles = fileStats.selectFiles(/.map$/, process.cwd());
+    SourceMapper.create(mapFiles, function(err3, mapper) {
+      if (err3) {
+        that.logger_.error('Error processing the sourcemaps.', err3);
```

Comment:
That would have inflated the patch without having any real meat to the change. The real fix would be to flatten that part of the code as well.

I think we should keep the variable as err3 to avoid unnecessary churn.